src/select.rs

```diff
@@ -6,13 +6,19 @@ use std::ascii::AsciiExt;
 use string_cache::{Atom, Namespace};
 use tree::{NodeRef, NodeData, ElementData};
 
+/// The definition of whitespace per CSS Selectors Level 3 § 4.
+///
+/// Copied from rust-selectors.
+static SELECTOR_WHITESPACE: &'static [char] = &[' ', '\t', '\n', '\r', '\x0C'];
+
+#[derive(Debug)]
 pub struct KuchikiSelectors;
 
 impl SelectorImpl for KuchikiSelectors {
     type NonTSPseudoClass = PseudoClass;
     type PseudoElement = PseudoElement;
```

Comment:
There are some typedefs missing here ([see selectors source](https://github.com/servo/rust-selectors/blob/cd0230444b1df96bfbdce509c0fab9f7f737c45d/src/parser.rs#L44)), but I couldn't figure out what those were good for, or what the entire trait was good for.